### ISSUE TITLE
Content is lost when converting nested lists

### ISSUE THREAD

**Issue by cyress** [akosbalasko/yarle#324]
If the exported `.enex` contains a nested list structure, the contents of the list elements that have sub-lists are no longer present after the conversion.

# Steps to reproduce
`npx -p yarle-evernote-to-md@latest yarle --configFile ./config.json`

## Configuration
_config.json_
```json
{
    "enexSources": [
       "/path/to/file/test.enex"
       ],
    "templateFile": "/path/to/file/template.tmpl",
    "outputDir": "/path/to/file/converted",
    "resourcesDir": "_resources",
    "isZettelkastenNeeded": false,
    "plainTextNotesOnly": false,    
    "useHashTags": false,
    "outputFormat": "ObsidianMD",
    "haveEnexLevelResources": false,
    "monospaceIsCodeBlock": false,
    "dateFormat": "DD.MM.YYYY hh:mm",
    "keepMDCharactersOfENNotes": false,
    "keepImageSize": "ObsidianMD",
    "urlEncodeFileNamesAndLinks": false,
    "skipEnexFileNameFromOutputPath": false,
    "keepOriginalAmountOfNewlines": false,
    "generateNakedUrls": true,
    "addExtensionToInternalLinks": true
}
```

## Input
_test.enex_
```xml
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE en-export SYSTEM "http://xml.evernote.com/pub/evernote-export3.dtd">
<en-export export-date="20210926T114156Z" application="Evernote Legacy" version="Evernote Mac 7.14.1 (458325)">
<note><title>Test note</title><content><![CDATA[<?xml version="1.0" encoding="UTF-8" standalone="no"?>
<!DOCTYPE en-note SYSTEM "http://xml.evernote.com/pub/enml2.dtd">
<en-note>
<ul>
    <li>
        List item 1
        <ul>
            <li>
                List item 1.1
                <ul>
                    <li>
                        List item 1.1.1
                    </li>
                </ul>
            </li>
            <li>
                List item 1.2
                <ul>
                    <li>
                        List item 1.2.1
                        <ul>
                            <li>
                                List item 1.2.1.1
                            </li>
                        </ul>
                    </li>
                </ul>
            </li>
            <li>
                List item 1.3
            </li>
        </ul>
    </li>
    <li>
        List item 2
        <ul>
            <li>
                List item 2.1
            </li>
        </ul>
    </li>
    <li>List item 3</li>
</ul>
</en-note>
]]></content><created>20160309T134711Z</created><updated>20160420T141949Z</updated><note-attributes></note-attributes></note>
</en-export>
```

## Output
_Test note.md_
```markdown
---
created-at: 09.03.2016 02:47
updated-at: 20.04.2016 04:19

---
*   List item 1.1.1

*   List item 1.2.1.1

*   List item 1.3

*   List item 2.1

*   List item 3
```

## Missing list-items
All list-items that have nested lists within get lost.
> List item 1
> List item 1.1
> List item 1.2
> List item 1.2.1
> List item 2

---
Environmental characteristics

* Yarle 4.5.3
* Node 16.8.0
* NPM 7.21.0

**Comment by scpedicini** [comment]
Just to add to this, it *seems* like empty div containers also end up "flattening" a nested list. I've noticed that evernote actually doesn't do this anymore, this is only on < v10 of evernote. Additionally, if you copy paste a bad table into a new note on 10, it also correctly strips out the empty div.

```
<div><b>Good List:</b></div>
<ul>
  <li>
    <div>Item 1</div>
  </li>
  <li>
    <div>Item 2</div>
  </li>
  <ul>
    <li>
      <div>Nested Item</div>
    </li>
    <li>
      <div>Nested Item</div>
    </li>
  </ul>
</ul>
```



```
<div><b>Bad List:</b></div>
<ul>
  <li>
    <div>Item 1</div>
  </li>
  <li>
    <div>Item 2</div>
  </li>
  <div>
  <ul>
    <li>
      <div>Nested Item</div>
    </li>
    <li>
      <div>Nested Item</div>
    </li>
  </ul>
  </div>
</ul>
```

The only difference is that the bad list as a div container surrounding the ul sublist.

**Comment by scpedicini** [comment]
Adding something like this to top of fixSublists seems to correct the issue.

```
const unwrapElement = (node: Element) => { node.replaceWith(...Array.from(node.children)); };

for (const ulNode of ulElements) {
  const parentNodeDiv = ulNode.parentElement;
  if (parentNodeDiv && parentNodeDiv.tagName === 'DIV') {
    unwrapElement(parentNodeDiv);
  }
}
```

Let me know if you'd like me to submit a PR.

**Comment by akosbalasko** [comment]
Hi @scpedicini , yes, please submit a PR. 
Thanks for your contribution!

**Comment by akosbalasko** [comment]
it looks @joaosa fixed this issue.

**Comment by cyress** [comment]
Hello @akosbalasko, Please excuse my late response.

I was about to test the fix, but if I am not mistaken, the Yarle version with the fix (4.8.3) is not available via NPM.
[Yarle's NPM details page](https://www.npmjs.com/package/yarle-evernote-to-md) says the latest version is 4.8.0 and when I run the `npx (...)@latest` command it only installs the 4.8.0 version.

It would be great if you could take a look at it.
Thanks you!

**Comment by akosbalasko** [comment]
Hi @cyress ,
I'm sorry, I simply missed to publish it. But now I did it, so it should work. Thanks for testing the fix!

**Comment by cyress** [comment]
Hello @akosbalasko,
i just wanted to ask if you might have had time to test/reproduce this behavior yet? ✌️
If my test was correct, the original issue is not resolved and the ticket should not be closed.

**Comment by cyress** [comment]
Hey @akosbalasko,
could you also test this again? Or should i copy the original error description and create a new issue therefor?

**Comment by akosbalasko** [comment]
Hi @cyress ,
Sorry for this long delay, I test it now.

**Comment by akosbalasko** [comment]
Hi @cyress ,
You're right, it looks like that the fix works only in case of the pretty-formatted lists like when <li> -tags are encapsulated closely, I mean there is no <ul> within. Check my unit tests for sublists here: https://github.com/akosbalasko/yarle/blob/master/test/data/test-sublists-multiple.enex
I repoen the issue and try to fix it. Thanks for raising this up again!

**Comment by akosbalasko** [comment]
@cyress , I released my fix in 4.9.13, please take a look at it. Thank you!
